### PR TITLE
Enhanced JSON format for data extraction in the docs

### PR DESCRIPTION
**Pull request by fady17**
This update enhances the JSON format for the extracted data, providing more structure and additional details. 

Changes include:
- Added meta description extraction.
- Included all page links.
- Added a content snippet for context.
- Improved error handling.

### CODE REVIEW

**Comment by fady17** [comment]
- Added meta description extraction.
- Included all page links.
- Added a content snippet for context.
- Improved error handling.

**Comment by vdusek** [comment]
@fady17 Please go through your changes once more. You completely broke the file.

**Comment by fady17** [comment]
sorry for the confusion i returned the main as it was and added a new example to the examples

File: docs/examples/extract-the-all-the-pagest.mdx [hunk 1]
```diff
@@ -0,0 +1,2713 @@
+i want you to work as an opensource contributer and start enhance the following code snipperts :import asyncio
```
Comment (B4nan) [inline docs/examples/extract-the-all-the-pagest.mdx:1]: this is clearly a "chatgpt rewrite", we are not interested in such PRs, if we want to, we can use it ourselves.

**Comment by fady17** [comment]
So nice to hear from you for future improvements thank you , but i just
what to highlight that it’s very clear I wasn’t try to hide chatgpt work
it’s a line in a tutorial for a vanilla feature to extract all the meta
deta  associated with the website instead of just extract the main one so I
didn’t think it’s write to write the whole tutorial back,
But you are absolutely correct and thanks for the advice.

**Pull request closed without merging**